testutils/utils.go

```diff
@@ -523,20 +523,27 @@ func testList(t *testing.T, kv store.Store) {
 		pairs, err := kv.List(parent)
 		assert.NoError(t, err)
 		if assert.NotNil(t, pairs) {
-			assert.Equal(t, len(pairs), 2)
+			assert.Equal(t, 2, len(pairs))
 		}
 
 		// Check pairs, those are not necessarily in Put order
 		for _, pair := range pairs {
 			if pair.Key == firstKey {
-				assert.Equal(t, pair.Value, firstValue)
+				assert.Equal(t, firstValue, pair.Value)
 			}
 			if pair.Key == secondKey {
-				assert.Equal(t, pair.Value, secondValue)
+				assert.Equal(t, secondValue, pair.Value)
 			}
 		}
 	}
-
+	// List should work and return 0 value
+	for _, key := range []string{firstKey, secondKey} {
+		pairs, err := kv.List(key)
+		assert.NoError(t, err)
+		if assert.NotNil(t, pairs) {
+			assert.Equal(t, 0, len(pairs))
+		}
+	}
```

Comment:
We should add a test case where the list is non-empty with a few elements and check we don't return the directory key on `List`? We should test for all cases (otherwise we would have missed the bad case for boltdb).

Reply:
@abronan OK, I'll crate the parent key in the `testList()` function.
In this way, 3 keys will be inserted (1 parent and 2 children) but only 2 Pairs have to been returned.